src/module/data/message/ability-use.mjs

```diff
@@ -87,5 +109,35 @@ export default class AbilityUseModel extends BaseMessageModel {
   /** @inheritdoc */
   addListeners(html) {
     super.addListeners(html);
+    /** @type {HTMLButtonElement[]} */
+    const effectButtons = html.querySelectorAll(".apply-effect");
+    for (const effectButton of effectButtons) effectButton.addEventListener("click", async (event) => {
+      /** @type {AppliedPowerRollEffect} */
+      const pre = await fromUuid(effectButton.dataset.uuid);
+      if (!pre) return void ui.notifications.error("DRAW_STEEL.Messages.AbilityUse.NoPRE", { localize: true });
+      const effectId = effectButton.dataset.effectId;
+      const config = pre.applied[this.tierKey].effects[effectId];
+
+      const noStack = !config.properties.has("stacking");
+
+      /** @type {DrawSteelActiveEffect} */
+      const tempEffect = effectButton.dataset.type === "custom" ?
+        pre.item.effects.get(effectId).clone({}, { keepId: noStack, addSource: true }) :
+        await DrawSteelActiveEffect.fromStatusEffect(effectId);
+
+      /** @type {ActiveEffectData} */
+      const updates = {
+        transfer: true,
+        origin: pre.uuid,
+        system: {},
+      };
+      if (config.end) updates.system.end = { type: config.end };
+      tempEffect.updateSource(updates);
+
+      for (const actor of ds.utils.tokensToActors()) {
+        // reusing the ID will block creation if it's already on the actor
+        actor.createEmbeddedDocuments("ActiveEffect", [tempEffect.toObject()], { keepId: noStack });
```

Comment:
I was unable to find the core issue, but yknow the one about being able to make only a single db operation - let's leave a comment here to improve this in the future (hopefully v14)